oak/server/grpc_event.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2018 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OAK_GRPC_SERVER_GRPC_EVENT_H_
+#define OAK_GRPC_SERVER_GRPC_EVENT_H_
+
+#include "grpc_stream.h"
+
+namespace oak {
+namespace grpc_server {
+
+class GrpcEvent {
+  public:
+  enum EventType {
```

Comment:
How about an alternative approach in which we use an abstract base class with a single method, and then virtually override that for the various cases we need to handle? So effectively rely on virtual dispatch instead of determining what to do based on an enum value.

e.g. (note: inaccurate syntax)

```
class BaseGrpcEvent {
  virtual void handle() = 0;
}

class ReadRequestEvent : BaseGrpcEvent {
  ...
}

class WriteResponseEvent : BaseGrpcEvent {
  ...
}
```

WDYT?

Reply:
I agree and updated the flow and I think it is nicer this way.